### PR TITLE
Check if base patches is beatable with all pickups

### PR DESCRIPTION
**Pull request by henriquegemignani**
Wild first version, not sure how decent it is. Seemed to work?

### CODE REVIEW

**Review by ThanatosGit — CHANGES_REQUESTED** [review]
Let me try to understand and learn the logic here.
After randomisation of elevators, starting area and probably some stuff I forgot about:
1. Resolver is used to check if the game is beatable with every item that will be placed + starting items.
2. If that's true the generator does the job and places the pickups.
3. Validation after generation would still be required because 2. could have placed an item unreachable which was required by 1.

But 1. (so this pr) would remove garbage where e.g. every elevator in a region leads to the same region, such that you could never beat the game.

File: randovania/game_description/game_patches.py [hunk 1]
```diff
@@ -5,6 +5,9 @@
 import typing
 from dataclasses import dataclass
 
+from randovania.game_description.assignment import (
+    PickupTarget,
+)
```
[the remark this answers is not in the document]
Comment (henriquegemignani) [reply randovania/game_description/game_patches.py:10]: At this point whatever ruff is fine with imports is fine for me.

**Comment by henriquegemignani** [comment]
> Let me try to understand and learn the logic here. After randomisation of elevators, starting area and probably some stuff I forgot about:

Mostly these two, but a game can shuffle more if they want. Echoes can do translator gates and portals.

>     1. Resolver is used to check if the game is beatable with every item that will be placed + starting items.
> 
>     2. If that's true the generator does the job and places the pickups.
> 
>     3. Validation after generation would still be required because 2. could have placed an item unreachable which was required by 1.
> 
> 
> But 1. (so this pr) would remove garbage where e.g. every elevator in a region leads to the same region, such that you could never beat the game.

Correct. 

I also wanted to catch starting at bad locations, but I don't think this does it. Some sort of baby generator where it checks how much you can reach with no pickups plus additional starting pickups would be needed.

File: randovania/game_description/game_patches.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ThanatosGit) [reply randovania/game_description/game_patches.py:10]: Then go for it :p

**Review by ThanatosGit — APPROVED** [review]

**Comment by henriquegemignani** [comment]
Will merge after 6.1

**Comment by ThanatosGit** [comment]
Oh, the merge failed.

**Pull request merged**